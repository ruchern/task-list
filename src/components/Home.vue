<template>
  <section class="hero is-fullheight">
    <div class="hero-body">
      <div class="container">
        <div class="columns">
          <div class="column is-half is-offset-one-quarter">
            <transition enter-active-class="animated flipInX">
              <Notification
                :key="message.content"
                :message="message"
                v-if="Object.keys(message).length > 0"
              />
            </transition>

            <div class="card">
              <SearchTask />

              <TaskList />

              <TaskFooter />
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</template>

<script>
import { mapState } from 'vuex'
import Notification from '@/components/Notification'
import SearchTask from '@/components/SearchTask'
import TaskList from '@/components/TaskList'
import TaskFooter from '@/components/TaskFooter'

export default {
  name: 'Home',
  components: {
    Notification,
    SearchTask,
    TaskList,
    TaskFooter
  },
  computed: {
    ...mapState(['message'])
  }
}
</script>

<style lang="scss" scoped>
.card {
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.2), 0 25px 50px 0 rgba(0, 0, 0, 0.1);
}

.hero {
  padding-top: 50px;
}
</style>
