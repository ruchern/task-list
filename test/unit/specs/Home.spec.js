import { shallow } from '@vue/test-utils'
import Home from '@/components/Home'

describe('Home', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(Home)
  })

  test('should render correct title', () => {
    expect(wrapper.find('h1').text()).toBe('Home')
  })
})
