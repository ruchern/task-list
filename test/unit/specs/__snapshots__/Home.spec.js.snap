// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render the component 1`] = `
<section
  class="hero is-fullheight"
>
  <div
    class="hero-body"
  >
    <div
      class="container"
    >
      <div
        class="columns"
      >
        <div
          class="column is-half is-offset-one-quarter"
        >
          <!---->
           
          <div
            class="card"
          >
            <searchtask-stub />
             
            <tasklist-stub />
             
            <taskfooter-stub />
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
